integration/containerd/cri/integration-tests.sh

```diff
@@ -129,18 +130,35 @@ create_containerd_config() {
 	# Remove dots.  Dots are used by toml syntax as atribute separator
 	runtime="${runtime//./-}"
 
-	cat << EOT | sudo tee "${CONTAINERD_CONFIG_FILE}"
+	if [ $kata_annotations -eq 1 ]; then
+		cat << EOT | sudo tee "${CONTAINERD_CONFIG_FILE}"
```

Comment:
The header of the file (until the annotation sections) and the footer (from the annotation on) are repeated in the `else` statement. Maybe you could do:
 1. Print the header
 2. Print the annotation sections if `$kata_annotations` equals to 1
 3. Print the footer
 
An alternative is to embedded the logic to print the annotation sections inside the cat's heredoc. For example:
```
cat << EOT | sudo tee "${CONTAINERD_CONFIG_FILE}"
[plugins]
  [plugins.cri]
    [plugins.cri.containerd]
        default_runtime_name = "$runtime"
      [plugins.cri.containerd.runtimes.${runtime}]
        runtime_type = "${runtime_type}"
        $( [ $kata_annotations -eq 1 ] && \
        echo 'pod_annotations = ["io.katacontainers.*"]' && \
        echo '        container_annotations = ["io.katacontainers.*"]'
        )
        [plugins.cri.containerd.runtimes.${runtime}.options]
          Runtime = "${containerd_runtime}"
[plugins.linux]
       shim = "${containerd_shim_path}"
EOT
```
The above command will produce a blank line if `$kata_annotations` equals to 0 though. Don't know if it will cause a problem. Also I confess that solution seems a bit ugly to me, maybe there should exist a best "bashism" way.

Reply:
Updated.

Reply:
Sorry to bother you, but we can use "echo -n" to avoid to produce a blank line. fix in #3943  :smile: